kore/src/Kore/Step/Simplification/Or.lhs

```diff
@@ -189,47 +187,18 @@ disjoinPredicates
         , predicate = predicate1
         , substitution = substitution1
         }
-    predicated2@Predicated
+    Predicated
         { term = term2
         , predicate = predicate2
         , substitution = substitution2
         }
 
-  | term1         == term2
-  =
-    let result
-          | substitution1 == substitution2 =
-            predicated1 { predicate = makeOrPredicate predicate1 predicate2 }
-\end{code}
-
-When the configurations have different substitutions, it is possible to disjoin
-them by promoting the substitutions into the predicate,
-```
-(t, p₁     , s₁) ∨ (t, p₂     , s₂)
-(t, p₁ ∧ s₁, ⊤ ) ∨ (t, p₂ ∧ s₂, ⊤ )
-```
-and following the algorithm described above for disjoining predicates when the
-substitutions are equal. This is not strictly a simplification: simplifying
-predicates extracts substitutions into the corresponding field of the
-configuration. Nevertheless, this simplification is required by
-`Kore.Step.Simplification.Equals.makeEvaluateFunctionalOr`.
-
-\begin{code}
-          | otherwise =
-            Predicated
-                { term = term1
-                , predicate =
-                    Function.on
-                        makeOrPredicate
-                        Predicated.toPredicate
-                        predicated1
-                        predicated2
-                , substitution = mempty
-                }
-    in Just (result, SimplificationProof)
-
+  | term1 == term2, substitution1 == substitution2 =
+    Just (result, SimplificationProof)
```

Comment:
Why not make an `or` of predicates when the substitutions are equal, but the predicates are different?

Reply:
That's exactly what this does. The change is: it no longer tries to combine predicates when the substitutions are inequal.